lib/middleware.js

```diff
@@ -508,19 +509,20 @@ module.exports = function middlewareConstructor() {
     },
 
     /**
-     * Check if a user exists for the given request and if a project
-     * has been loaded
+     * Check if if a project has been loaded for the current context
      */
-    isProjectLoaded: function(req, res, next) {
-      if(!req.session.user) {
-        return next(utils.error(401, "Unauthorized request"));
-      }
+    isProjectLoaded: function(allowAnonymous) {
+      return function(req, res, next) {
+        if(!req.session.user && !allowAnonymous) {
```

Comment:
This isn't great. We probably want a separate middleware for this piece. In fact, [we have one!](https://github.com/gideonthomas/thimble.webmaker.org/blob/remix/lib/middleware.js#L114) I think you should modify that one to allow anonymous instead of `isProjectLoaded`